### PR TITLE
feat: add AMQP 0-9-1 support

### PR DESCRIPTION
**Pull request by AceTheCreator**
This PR adds support for AMQP 0-9-1.

cc @fmvilas @Souvikns @KhudaDad414 

fixes: #257

### CODE REVIEW

**Review by KhudaDad414 — CHANGES_REQUESTED** [review]
Left a few comments.

File: src/adapters/amqp/index.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    const { url, auth, serverBindings, protocolVersion } = data
+
+    return amqplib.connect({
+      host: url.hostname,
+      port: url.port || 5672,
+      protocol: url.protocol.slice(0, url.protocol.length - 1),
+      username: auth.username,
+      password: auth.password,
+      keepalive: serverBindings?.keepAlive || 0,
+      vhost: serverBindings?.vhost || '/',
```
[the remark this answers is not in the document]
Comment (AceTheCreator) [reply src/adapters/amqp/index.ts:37]: Yea, we don't. I was playing around trying a default connection. I'll definitely get rid of it

File: src/adapters/amqp/index.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+          return ch.close()
+        })
+        })
+      }).finally(() => this.client.connection.close())
+    })
+  }
+  
+
+  async _connect(): Promise<this> {
+    const resolved = false
```
[the remark this answers is not in the document]
Comment (AceTheCreator) [reply src/adapters/amqp/index.ts:141]: You're right!

**Comment by AceTheCreator** [comment]
@KhudaDad414 i updated this PR, have a look

**Pull request closed without merging**